src/runtime_src/core/common/api/xrt_kernel.cpp

```diff
@@ -2330,6 +2343,11 @@ class run_impl
       xrt_core::module_int::sync(m_module);
 
     prep_start();
+    // log kernel start info
+    // This is in critical path, we need to reduce log overhead 
+    // as much as possible, passing kernel impl pointer instead of 
+    // constructing args in place
+    m_usage_logger->log_kernel_run_info(kernel.get(), this, true, ERT_CMD_STATE_NEW);
```

Comment:
I wonder if the bool argument to `log_kernel_run_info` is needed to could simply be inferred from `state != ERT_CMD_STATE_NEW` ?